pkg/gofr/datasource/solr/solr.go

```diff
@@ -0,0 +1,215 @@
+package solr
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+)
+
+type Config struct {
+	Host string
+	Port string
+}
+
+type Client struct {
+	url string
+
+	logger  Logger
+	metrics Metrics
+}
+
+// New initializes Solr driver with the provided configuration.
+// The Connect method must be called to establish a connection to Solr.
+// Usage:
+// client := New(config)
+// client.UseLogger(loggerInstance)
+// client.UseMetrics(metricsInstance)
+// client.Connect()
+func New(conf Config) *Client {
+	s := &Client{}
+	s.url = "http://" + conf.Host + ":" + conf.Port + "/solr/"
```

Comment:
I would remove the trailing slash here

```suggestion
	s.url = "http://" + conf.Host + ":" + conf.Port + "/solr"
```


And then add it back when handling API calls

See https://github.com/gofr-dev/gofr/pull/985#discussion_r1760613769